lib/rules/no-nested-landmark.js

```diff
@@ -0,0 +1,59 @@
+'use strict';
+
+const Rule = require('./base');
+const AstNodeInfo = require('../helpers/ast-node-info');
+
+function createErrorMessage(element) {
+  return `Nested landmark elements on <${element}> detected. Landmark elements should not be nested within landmark elements of the same name.`;
+}
+
+const LANDMARK_ELEMENTS = ['header', 'main', 'aside', 'form', 'main', 'nav', 'footer'];
+const ROLES = ['banner', 'main', 'complementary', 'form', 'search', 'navigation', 'contentinfo'];
+
+module.exports = class NoNestedLandmark extends Rule {
+  visitor() {
+    this._parentLandmarkNode = null;
+
+    let visitor = {
```

Comment:
It is vaguely confusing to me when reading a method named `vistor` with an inline variable named `visitor`. Maybe we can call this something else (e.g. `process`/`checkNode`,etc?).